logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/scenario/OnConfigRequest.kt

```diff
@@ -0,0 +1,43 @@
+package com.wire.kalium.logic.feature.call.scenario
+
+import com.sun.jna.Pointer
+import com.wire.kalium.calling.Calling
+import com.wire.kalium.calling.callbacks.CallConfigRequestHandler
+import com.wire.kalium.calling.types.Handle
+import com.wire.kalium.logic.callingLogger
+import com.wire.kalium.logic.data.call.CallRepository
+import com.wire.kalium.logic.feature.call.AvsCallBackError
+import com.wire.kalium.logic.feature.call.CallManagerImpl
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+//TODO create unit test
+class OnConfigRequest(
+    private val calling: Calling,
+    private val callRepository: CallRepository
+) : CallConfigRequestHandler {
+    override fun onConfigRequest(inst: Handle, arg: Pointer?): Int {
+        callingLogger.i("${CallManagerImpl.TAG} - onConfigRequest")
+
+        //TODO use the same coroutine job of call manager
+        CoroutineScope(Dispatchers.IO).launch {
+            val config = callRepository.getCallConfigResponse(limit = null)
+                .fold({
+                    TODO("")
+                }, {
+                    it
+                })
+
+            calling.wcall_config_update(
+                inst = inst,
+                error = 0, // TODO: http error from internal json
```

Comment:
What does this `http error form interl json` mean?

Reply:
This is the http error code we need to get from `callApi.getCallConfig(..)`, we will then need to map it from `CoreFailure` to something from `NetworkResponse.Error` and try to get the errorCode. Which is something I think we can leave for later, at least for now until we get most of the basics of calling in place.